### PR TITLE
PIM-5097: Add catalog import jobs

### PR DESCRIPTION
**Pull request by SamirBoulil**
This PR adds XLSX job import of : 

- Categories
- association types
- attributes
- options
- families

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | yes
| Changelog updated                 | yes
| Review and 2 GTM                  | yes
| Micro Demo to the PO (Story only) | yes
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: features/import/import_attributes.feature [hunk 1]
```diff
@@ -145,7 +145,7 @@ Feature: Import attributes
     And the following CSV file to import:
       """
       type;code;label-de_DE;label-en_US;label-fr_FR;group;unique;useable_as_grid_filter;allowed_extensions;metric_family;default_metric_unit;reference_data_name;localizable;scopable;available_locales;sort_order;max_characters;validation_rule;validation_regexp;wysiwyg_enabled;number_min;number_max;decimals_allowed;negative_allowed;date_min;date_max;metric_family;default_metric_unit;max_file_size;allowed_extensions
-      pim_catalog_simpleselect;manufacturer;Meine große Code;My awesome code;Mon super code;marketing;0;1;;;;;0;0;en_US,fr_FR;3;300;validation_rule;;1;3;5;true;true;2000-12-12;2015-08-08;;EUR;452;jpg
+      pim_catalog_simpleselect;manufacturer;Meine große Code;My awesome code;Mon super code;marketing;0;1;;;;;0;0;en_US,fr_FR;3;300;validation;;1;3;5;true;true;2000-12-12;2015-08-08;;EUR;452;jpg
```
Comment (SamirBoulil) [inline features/import/import_attributes.feature]: 'validation_rule' string exceeds the attribute length in DB that is 10

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/csv_connector.yml [hunk 2]
```diff
@@ -108,7 +108,7 @@ connector:
             type: import
             steps:
                 validation:
-                    title: pim_connector.jobs.csv_group_import.validation.title
+                    title: pim_connector.jobs.csv_association_type_import.validation.title
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/csv_connector.yml]: good catch :)

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml [hunk 3]
```diff
@@ -24,6 +24,83 @@ connector:
...
+                    services:
+                        reader:    pim_connector.reader.file.xlsx_attribute_option
+                        processor: pim_connector.processor.denormalization.attribute_option.flat
+                        writer:    pim_connector.writer.doctrine.attribute_option
+        xlsx_association_type_import:
+            title: pim_connector.jobs.xlsx_association_type_import.title
+            type: import
+            steps:
+                validation:
+                    title: pim_connector.jobs.xlsx_group_import.validation.title
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml]: should not be xlsx_group_import, but xlsx_association_type_import

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml [hunk 4]
```diff
@@ -24,6 +24,83 @@ connector:
...
+                    class: %pim_connector.step.validator.class%
+                    services:
+                        charsetValidator: pim_connector.validator.item.charset_validator
+                import:
+                    title:         pim_connector.jobs.xlsx_association_type_import.import.title
+                    services:
+                        reader:    pim_connector.reader.file.xlsx_association
```
Comment (skeleton) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml]: should be `pim_connector.reader.file.xlsx_association_type`

**Comment by juliensnz** [comment]
GTM

File: src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml [hunk 5]
```diff
@@ -109,3 +109,15 @@ services:
 
     pim_connector.reader.file.xlsx_variant_group:
         alias: pim_connector.reader.file.xlsx_product
+
+    pim_connector.reader.file.xlsx_attribute:
```
Comment (skeleton) [inline src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml]: missing `pim_connector.reader.file.xlsx_association_type` reader

File: src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml (hunk 3 above)
Comment (SamirBoulil) [inline src/Pim/Bundle/BaseConnectorBundle/Resources/config/batch_jobs/xlsx_connector.yml]: yes good catch! damn, I duplicated the error I picked up

**Comment by skeleton** [comment]
gtm once fixed

File: src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml (hunk 5 above)
Comment (SamirBoulil) [inline src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml]: it's here. It should be renamed. I agree https://github.com/akeneo/pim-community-dev/blob/PIM-5097/src/Pim/Bundle/ConnectorBundle/Resources/config/readers.yml#L104

**Pull request merged**